src/CSEjector.sol

```diff
@@ -78,42 +188,61 @@ contract CSEjector is
             revert AlreadyWithdrawn();
         }
 
-        uint256 pointer = _keyPointer(nodeOperatorId, keyIndex);
-        if (_isValidatorEjected[pointer]) {
-            revert AlreadyEjected();
-        }
-
         uint256 curveId = ACCOUNTING.getBondCurveId(nodeOperatorId);
 
-        ICSParametersRegistry registry = MODULE.PARAMETERS_REGISTRY();
-
-        (, uint256 threshold) = registry.getStrikesParams(curveId);
+        (, uint256 threshold) = PARAMETERS_REGISTRY.getStrikesParams(curveId);
         if (strikes < threshold) {
             revert NotEnoughStrikesToEject();
         }
 
-        uint256 penalty = registry.getBadPerformancePenalty(curveId);
-        if (penalty > 0) {
-            ACCOUNTING.penalize(nodeOperatorId, penalty);
-        }
-
         bytes memory pubkey = MODULE.getSigningKeys(
             nodeOperatorId,
             keyIndex,
             1
         );
+        uint256 penalty = PARAMETERS_REGISTRY.getBadPerformancePenalty(curveId);
+        bytes32 noPubkeyPacked = _nodeOperatorPublicKeyPacked(
+            nodeOperatorId,
+            pubkey
+        );
+        if (_exitPenaltyInfo[noPubkeyPacked].strikesPenalty.isValue) {
+            _exitPenaltyInfo[noPubkeyPacked].strikesPenalty = MarkedUint248(penalty.toUint248(), true);
+            emit BadPerformancePenaltyProcessed(
+                nodeOperatorId,
+                pubkey,
+                penalty.toUint248()
+            );
+        }
+
         // TODO: make the function payable and call `requestEjection{ value: msg.value }(pubkey)`
+    }
 
-        _isValidatorEjected[pointer] = true;
-        emit EjectionSubmitted(nodeOperatorId, keyIndex, pubkey);
+    /// @inheritdoc ICSEjector
+    function isValidatorExitDelayPenaltyApplicable(
+        uint256 nodeOperatorId,
+        bytes calldata publicKey,
+        uint256 eligibleToExitInSec
+    ) external view onlyCSM returns (bool)  {
+        uint256 curveId = ACCOUNTING.getBondCurveId(nodeOperatorId);
+        uint256 allowedExitDelay = PARAMETERS_REGISTRY.getAllowedExitDelay(
+            curveId
+        );
+        if (eligibleToExitInSec < allowedExitDelay) {
+            return false;
+        }
+        return !_exitPenaltyInfo[_nodeOperatorPublicKeyPacked(nodeOperatorId, publicKey)].delayPenalty.isValue;
```

Comment:
We need to think about the probabilities of the conditions here. The most likely one should be placed first. I doubt there will be on-chain calls with insufficient delay. Hence, lets rearrange the conditions

Reply:
don't get the point tbh
this method is for offchain mostly, and you can't exit early when there is no penalty value recorded, because the delay is not here yet